modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,906 @@
+    */
+    bool buildRegtree(regtree &tree,vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
+    /** @brief This function greedily decides the best split among the test splits generated.
+    * @param samples vector storing the training samples
+    * @param pixel_coordinates vector for storing the random test coordinates generated..
+    * @param start stores start index of the samples currently at the node
+    * @param end stores end index of the samples currently at the node
+    */
+    splitr generateSplit(vector<Point2f> pixel_coordinates, vector<training_sample> samples,unsigned long start ,unsigned long end);
+    /**@brief This function randomly generates test coordinates for each level of cascade.
+    * @param pixel_coordinates stores the test coordinates generated for making splits.
+    * @param min_x minimum x coordinate of the random pixels generated
+    * @param max_x maximum x coordinate of the random pixels generated
+    * @param min_y minimum y coordinate of the random pixels generated
+    * @param max_y maximum y coordinate of the random pixels generated
+    */
+    void getTestCoordinates (vector< vector<Point2f> >& pixel_coordinates,float min_x,
+                                        float min_y, float max_x , float max_y);
+    /** @brief This function gets the landmarks in the meanshape nearest to the pixel coordinates.
+    * @param pixels stores the point nearest to which shape coorinates are to be determined.
+    */
+    unsigned long getNearestLandmark (Point2f pixels );
+    /** @brief This function implements gradient boosting by fitting regression trees
+    * to get weak learners and update the shape residual.
+    *@returns A forest of regression trees fitted which act as weak learners for the model.
+    *@param samples stores the training samples.
+    *@param pixel_coordinates stores the test pixel coordinates for making the split.
+    */
+    vector<regtree> gradientBoosting(vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
+    /**@brief This function creates training sample by randomly assigning a current shape from set of
+    * shapes available.
+    *@returns a bool value
+    *@param samples An empty vector that would hold training samples generated.
+    *@param images vector which stores the images which are available as training data.
+    *@param landmarks vector which stores the actual shape of each corresponding image which is available as training data.
+    */
+    bool createTrainingSamples(vector<training_sample> &samples,vector<Mat> images,vector< vector<Point2f> > landmarks);
+};
+
+FacemarkKazemi::~FacemarkKazemi(){};
+
+bool FacemarkKazemiimpl:: getFaces(Mat src,vector<Rect> &facep,int select,double scaleFactor
+        , int minNeighbors, int flags, Size minSize, Size maxSize){
+    vector<Rect> facesp;
+    string face_cascade_name;
+    switch(select){
+        case FACE_HAAR : face_cascade_name = "haarcascade_frontalface_alt.xml" ;
```

Comment:
I am unsure how we can force OpenCV modules to look at the location of these files hardcoded. Paths, especially relative ones are always dangerous in source code. @alalek could you maybe suggest the best way to deal with using different xml files in the backend source code?